spot-client/src/common/remote-control/BaseRemoteControlService.js

```diff
@@ -138,6 +150,62 @@ export class BaseRemoteControlService extends Emitter {
             });
     }
 
+    /**
+     * Creates a P2P signaling connection that if successfully established will be used to send/receive remote control
+     * commands.
+     *
+     * @param {boolean} isServer - Whether to create a server or a client P2P signaling connection.
+     * @protected
+     * @returns {void}
+     */
+    _createP2PSignalingConnection(isServer) {
+        logger.log('Will try to establish P2P signaling channel', { isServer });
+
+        const P2PSignalingType = isServer ? P2PSignalingServer : P2PSignalingClient;
```

Comment:
I don't have clear idea on this one, because I don't know how to get rid of duplication in case each subclass is creating it. Would `getP2PSignalingType` abstract method be okay with you?